recipe/meta.yaml

```diff
@@ -1,52 +1,125 @@
-{% set name = "ariadne-codegen" %}
-{% set version = "0.13.0" %}
+{% set version = "0.14.0" %}
+
+# handle undefined PYTHON in `noarch: generic` outputs
+{% if PYTHON is not defined %}{% set PYTHON = "$PYTHON" %}{% endif %}
 
 package:
-  name: {{ name|lower }}
+  name: ariadne-codegen
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ariadne_codegen-{{ version }}.tar.gz
-  sha256: 52aa67037dcd88a2f6013118e69ba0ed35a49f37312668a1969907f0e9c9bf8b
+  - folder: dist
+    url: https://pypi.io/packages/source/a/ariadne-codegen/ariadne_codegen-{{ version }}.tar.gz
+    sha256: d5b15470195c3858b88a2bae855b5c04602ffae40228077b6cfb26a57d3f14f6
+  - folder: src
+    url: https://github.com/mirumee/ariadne-codegen/archive/refs/tags/{{ version }}.tar.gz
+    sha256: 86d79970c8260fa8292e40ada695babcf47560d6ac599611d039daedaae8543c
 
 build:
+  number: 0
+  noarch: python
+  script: cd dist && {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
   entry_points:
     - ariadne-codegen = ariadne_codegen.main:main
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  number: 0
 
 requirements:
   host:
-    - python >=3.7
     - hatchling
     - pip
+    - python >=3.9
   run:
-    - python >=3.7
-    - click >=8.1,<9.dev0
-    - graphql-core >=3.2.0,<3.3
-    - toml >=0.10,<1.dev0
-    - httpx >=0.23,<1.dev0
-    - pydantic >=2.0.0,<3.0.0
+    - autoflake
     - black
+    - click >=8.1,<9.0
+    - graphql-core >=3.2.0,<3.3
+    - httpx >=0.23,<1.0
     - isort
-    - autoflake
+    - pydantic >=2.0.0,<3.0.0
+    - python >=3.9
+    - toml >=0.10,<1.0
+  run_constrained:
+    - websockets >=11.0,<12.0
 
 test:
+  requires:
+    - pip
   imports:
     - ariadne_codegen
   commands:
     - pip check
     - ariadne-codegen --help
-  requires:
-    - pip
+
+outputs:
+  - name: ariadne-codegen
+  - name: ariadne-codegen-with-websockets
+    build:
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage("ariadne-codegen", max_pin="x.x.x") }}
```

Comment:
I think we want `exact=True` here instead, also below

Reply:
`exact` matches down to the build hash... the hashes will never match between a `noarch: generic` and `noarch: python`

Reply:
Works in #3 though